### PR TITLE
Create General Release Guidance and Put it in the Outbound Section

### PR DESCRIPTION
**Pull request by IsaacMilarky**
## Problem

We don't have general release guidance anywhere.

## Solution

Adapt the [release-guidelines-template](https://github.com/DSACMS/repo-scaffolder/blob/main/release-guidelines-template.md) into a general document.

## Result

Added release-guidelines.md

### CODE REVIEW

**Review by natalialuzuriaga — APPROVED** [review]
LGTM! But I'm not sure if we want a direct copy of the file here or instead point at the one in repo-scaffolder, I'll defer to Remy

**Review by decause-gov — APPROVED** [review]
LGTM +1 🚢 🌳 

IDEA: Track a list of the gh actions we're depending on
Q: Should we link to this in the guidance in the `MAINTAINERS.md` in repo-scaffolder?
A: yes! Then we don't maintain 3 versions for each tier!

https://github.com/DSACMS/repo-scaffolder/blob/main/tier3/%7B%7Bcookiecutter.project_slug%7D%7D/MAINTAINERS.md

**Pull request merged**
